[Manual Requests] Create thread for Manual Request when opening IOUPreview

bondydaa opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

This will create a thread for a Manual Request if one does not exist yet when a user click to open a Manual Request. 

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/Expensify/issues/278887
$ https://github.com/Expensify/Expensify/issues/270591


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

1. As userA create a chat with userB
2. Request money from userB - $10 for lunch
3. Click on the IOUPreview
4. Make sure you have been navigated to a new chat report
5. Verify you got a new report in LHN with title: `$10 request for lunch`
6. Send a message in this report
7. As userB verify you got this report in your LHN
8. Got to the DM with UserA as userB
9. Verify the IOUPreview has replies UI underneath like so 
<img width="501" alt="image" src="https://github.com/Expensify/App/assets/36083550/9780c9bf-648e-44ac-b55d-f3f91b6cceee">

10. Click on the IOUPreview for the $10 request
11. Verify you are navigated to the same thread created after UserA sent a message in step 6

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
12. Upload an image via copy paste
13. Verify a modal appears displaying a preview of that image
--->

Same as tests

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/36083550/80e3a907-4702-4c61-9dea-90b2364c5d54


</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

Cant sign in on mWeb android locally for some reason, dont want to hold on this though tried to figure out but its happening on main too

</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/36083550/50e7e6c1-79ac-4109-bf3c-f704605e56be


</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/36083550/28b64ec0-0dec-42c1-84bb-167beb37c7fc


</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/36083550/4328ba92-b8ee-4e54-bc6a-5f44ca13c5aa


</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

https://github.com/Expensify/App/assets/36083550/b54be6c6-c2ed-4ad8-acf2-37fcf1d0bae9



</details>

bondydaa committed:
create method to find if a thread exists or not

bondydaa committed:
add rough sketch of what needs to happen to create a thread on demand

Julesssss commented on src/components/ReportActionItem/IOUAction.js:
```diff
@@ -71,6 +72,10 @@ const defaultProps = {
     const onIOUPreviewPressed = () => {
+        if (!doesThreadExistForReportActionID(props.action.reportActionID)) {
+            // optimistically create reportID
+            // pass it along to OpenReport since that is build to create a thread when needed, not sure what to do about the navigation stuff below...
```
The navigation logic is likely going to change, I wouldn't worry too much just yet.

bondydaa replied:
okay coolio thanks Jules!

bondydaa committed:
fixing merge conflict

bondydaa committed:
rename method and update to return the report object for a thread

bondydaa committed:
allow the passing of a parentReportActionID so we can create threads

bondydaa committed:
generate report optimistically and pass parentReportActionID to create a thread

bondydaa commented:
Okay getting closer. I've got the optimistic report stuff done, via the logs I (accidentally) committed you can see the object has the proper data 
<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/346da3cd-d19c-4237-8f44-93f8c99bd433">

But I need to look closer at what's happening when that gets to the API and Auth b/c right now when I look up the report in the db for this it's not right:
```
sqlite> select * from reports where reportid=5450562980361980;
   created = 2023-05-12 05:07:27
 submitted =
  approved =
  reportID = 5450562980361980
 accountID = 0
 managerID =
reportName = Chat Report
     state = 0
     total = 0
  currency = USD
cachedData =
    status = 0
```

that good news 🎉 is that the rNVP is being created properly :success:
```
sqlite> select * from reportNameValuePairs where value=3983353111135864060;
reportID = 5450562980361980
    name = parentReportActionID
   value = 3983353111135864060
```

luacmartins commented on src/components/ReportActionItem/MoneyRequestAction.js:
```diff
@@ -15,6 +15,10 @@ import Navigation from '../../libs/Navigation/Navigation';
 import ROUTES from '../../ROUTES';
 import styles from '../../styles/styles';
 import * as IOUUtils from '../../libs/IOUUtils';
+import {getThreadForReportActionID, buildOptimisticChatReport} from '../../libs/ReportUtils';
```
We should import the library and then call each specific method, e.g. `ReportUtils.getThreadForReportActionID`

```suggestion
import * as ReportUtils from '../../libs/ReportUtils';
```

luacmartins commented on src/components/ReportActionItem/MoneyRequestAction.js:
```diff
@@ -15,6 +15,10 @@ import Navigation from '../../libs/Navigation/Navigation';
+import {getThreadForReportActionID, buildOptimisticChatReport} from '../../libs/ReportUtils';
+import * as Report from '../../libs/actions/Report';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import lodashGet from 'lodash/get';
```
lint

luacmartins commented on src/components/ReportActionItem/MoneyRequestAction.js:
```diff
@@ -66,11 +78,40 @@ const defaultProps = {
+                '',
+                props.chatReport.policyID,
+                props.chatReport.owner,
+                props.chatReport.type === CONST.REPORT.CHAT_TYPE.POLICY_EXPENSE_CHAT,
```
This should be `props.chatReport.isOwnPolicyExpenseReport`, no?

bondydaa replied:
ah yeah makes sense

luacmartins commented on src/components/ReportActionItem/MoneyRequestAction.js:
```diff
@@ -66,11 +78,40 @@ const defaultProps = {
+            console.log({optimisticThreadReport});
+            Report.openReport(
+                optimisticThreadReport.reportID,
+                [_.reject(optimisticThreadReport.participants, (login) => login === lodashGet(props.session, 'email', ''))],
```
Should this be the same as `props.chatReport.participants`?

bondydaa replied:
I wasn't 100% sure since the API param for participants says it shouldn't include the user making the request: https://github.com/Expensify/Web-Expensify/blob/335001f0f64bf449fc884456ef1911cd1e6f17d5/lib/ReportAPI.php#L4404-L4405

bondydaa replied:
okay asked about this here https://expensify.slack.com/archives/C02HWMSMZEC/p1683912289696679?thread_ts=1683911072.385549&cid=C02HWMSMZEC

And looks like it doesn't really matter if we include the user making the request here because in Auth here we make sure to include them and since `set` doesn't allow duplicates we are fine https://github.com/Expensify/Auth/blob/dbd779f7c423a82d4871d6dc076ecb37406fb43c/auth/command/OpenReport.cpp#L77

luacmartins commented on src/languages/en.js:
```diff
@@ -328,6 +328,7 @@ export default {
         payerOwesAmount: ({payer, amount}) => `${payer} owes ${amount}`,
         noReimbursableExpenses: 'This report has an invalid amount',
         pendingConversionMessage: "Total will update when you're back online",
+        threadReportName: ({payee, comment}) => `${payee} request${comment ? ` for ${comment}` : ''}`,
```
This should be `<formattedAmount> request for <comment>`

bondydaa replied:
haha gotcha I wasn't 100% sure based on the doc b/c I'd copied it as `$X request for $comment` and wasn't sure if X was meant to be amount or the person making the request 😂

luacmartins commented on src/libs/ReportUtils.js:
```diff
@@ -1276,6 +1276,7 @@ function buildOptimisticChatReport(
     oldPolicyName = '',
     visibility = undefined,
     notificationPreference = CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS,
+    parentReportActionID = '',
```
Add JSDoc for `parentReportActionID`

luacmartins reviewed (commented):
Looking good. Just a few minor comments.

bondydaa committed:
Merge branch 'main' of github.com:Expensify/App into bondy-thread-manual-requests

bondydaa committed:
appease linter

bondydaa committed:
appease linter and import full lib

bondydaa committed:
we dont need to prefilter the user making the request out since auth will do this for us

bondydaa committed:
use amount not payee in report name

bondydaa committed:
fixing jsdocs and comment

bondydaa commented on src/languages/en.js:
```diff
@@ -329,6 +329,7 @@ export default {
         payerSettled: ({amount}) => `settled up ${amount}`,
         noReimbursableExpenses: 'This report has an invalid amount',
         pendingConversionMessage: "Total will update when you're back online",
+        threadReportName: ({formattedAmount, comment}) => `${formattedAmount} request${comment ? ` for ${comment}` : ''}`,
```
Should be coming through now
<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/4073354/467dbbc4-590e-4595-9f3b-3788a15c8aad">

bondydaa committed:
use proper prop

bondydaa commented:
Okay I addressed all the comments but for some reason I'm still unsure why the optimistic report object that is being passed to OpenReport isn't leading to the data in the DB being set properly.

```
sqlite> select * from reports where reportid=2326627911912510;
   created = 2023-05-12 18:52:46
 submitted =
  approved =
  reportID = 2326627911912510
 accountID = 0
 managerID =
reportName = Chat Report
     state = 0
     total = 0
  currency = USD
cachedData =
    status = 0
```


The thread does get created though which is our main goal here 

```
sqlite> select * from reportNameValuePairs where value=3983353111135864060;
reportID = 2326627911912510
    name = parentReportActionID
   value = 3983353111135864060
```

when testing this I've had to manually delete the data from the DB and also out of the localDB by searching `report_[reportID]` and clicking delete otherwise the front end still thinks a thread already exists. 

I have to sign off for now so if someone can take this over that would be great. I don't expect to be online Monday but if I am able to then I'll let people know.

bondydaa commented on src/components/ReportActionItem/MoneyRequestAction.js:
```diff
@@ -66,11 +79,39 @@ const defaultProps = {
+            Report.openReport(optimisticThreadReport.reportID, optimisticThreadReport.participants, optimisticThreadReport);
+        }
         if (hasMultipleParticipants) {
             Navigation.navigate(ROUTES.getReportParticipantsRoute(props.chatReportID));
```
Note that I think in order to get the newly created thread to show up we'll need to have something like GetChats triggered OR navigate to the report here in this section. 

@Julesssss and @luacmartins had mentioned that this might be in the works else where as well but I'm not sure of a GH or anything for it. 

If nothing changes here with these routes then the modal will continue to open instead of going to the thread.

mountiny committed:
Merge main and resolve conflicts

mountiny committed:
Fix linter

mountiny committed:
Merge branch 'main' into bondy-thread-manual-requests

mountiny committed:
Merge branch 'vit-ensureDeleteShowsAsText' into bondy-thread-manual-requests

mountiny committed:
Ensure threads are created optimistically

mountiny committed:
Add the childReportID to the report action so we do not create multiple reports

mountiny committed:
Merge branch 'main' into bondy-thread-manual-requests

mountiny commented on src/components/ReportActionItem/MoneyRequestAction.js:
```diff
@@ -67,15 +80,46 @@ const defaultProps = {
     const hasMultipleParticipants = lodashGet(props.chatReport, 'participants', []).length > 1;
     const onIOUPreviewPressed = () => {
-        if (hasMultipleParticipants) {
+        if (lodashGet(props.action, 'originalMessage.type', '') === CONST.IOU.REPORT_ACTION_TYPE.SPLIT && hasMultipleParticipants) {
```
We only going to navigate to the details page in case of IOU split preview

mountiny commented on src/components/ReportActionItem/MoneyRequestAction.js:
```diff
@@ -67,15 +80,46 @@ const defaultProps = {
+        // If the childReportID is not present, we need to create a new thread
+        const childReportID = lodashGet(props.action, 'childReportID', '0');
+        if (childReportID === '0') {
+            const participants = _.uniq([props.session.email, props.action.actorEmail]);
+            const formattedUserLogins = _.map(participants, (login) => OptionsListUtils.addSMSDomainIfPhoneNumber(login).toLowerCase());
+            const thread = ReportUtils.buildOptimisticChatReport(
+                formattedUserLogins,
+                props.translate('iou.threadReportName', {
+                    formattedAmount: CurrencyUtils.convertToDisplayString(lodashGet(props.action, 'originalMessage.amount', 0), lodashGet(props.action, 'originalMessage.currency', '')),
+                    comment: props.action.originalMessage.comment,
+                }),
+                '',
+                CONST.POLICY.OWNER_EMAIL_FAKE,
+                CONST.POLICY.OWNER_EMAIL_FAKE,
+                false,
+                '',
+                undefined,
+                CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS,
+                props.action.reportActionID,
+                props.chatReportID, // Needs to be changed to iouReportID
+            );
```
Eventually we should move this to reportUtils but now for urgency just leaving this pragmatically here, I hope that works

luacmartins replied:
Yea, we should create an utils method. Seems ok to leave it for now given that this is blocking other issues

mountiny commented on src/libs/ReportUtils.js:
```diff
@@ -1011,6 +1024,9 @@ function getReportName(report) {
     let formattedName;
     if (isThread(report)) {
         const parentReportAction = ReportActionsUtils.getParentReportAction(report);
+        if (ReportActionsUtils.isTransactionThread(parentReportAction)) {
+            return getTransactionReportName(parentReportAction);
```
The reportName is specific for the transaction threads so hijackign this flow

mountiny committed:
Remove unsused method

mountiny commented on src/libs/actions/Report.js:
```diff
@@ -404,6 +404,20 @@ function openReport(reportID, participantList = [], newReportObject = {}, parent
         }
+
+        // If we are creating a thread, ensure the report action has childReportID property added
+        if (newReportObject.parentReportID && parentReportActionID) {
+            onyxData.optimisticData.push({
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${newReportObject.parentReportID}`,
+                value: {[parentReportActionID]: {childReportID: reportID}},
+            });
+            onyxData.failureData.push({
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${newReportObject.parentReportID}`,
+                value: {[parentReportActionID]: {childReportID: '0'}},
+            });
+        }
```
This makes sure that when we create a report, the parent report action gets the childReportID key added. It should be updated by pusher, but if you are offline we would keep creating new threads by clicking on the preview or we could also crete new ones by quickly clicking on the preview before the pusher update comes through

mountiny replied:
I ma not sure but I think no successData is needed here because we are only adding a property in ptimistic data and no pending value

mountiny commented on src/pages/home/report/ReportActionItemParentAction.js:
```diff
@@ -41,6 +42,11 @@ const defaultProps = {
 const ReportActionItemParentAction = (props) => {
     const parentReportAction = props.parentReportActions[`${props.report.parentReportActionID}`];
+
+    // In case of transaction threads, we do not want to render the parent report action.
+    if (ReportActionsUtils.isTransactionThread(parentReportAction)) {
+        return null;
+    }
```
This is to ensure we do not show any report Action at the top of the transactions thread

grgia replied:
I'm working on a change that might affect this line, can you clarify what the goal here was? @mountiny

fedirjh reviewed (commented):
The tests are failing for me , the report is created then deleted. 
There is a bug in the backend , the `parentReportActionID` is rounded in server. This discrepancy caused some bugs.

The `parentReportActionID` which is created optimistically is `1532330860146572254` 

<details><summary>Screenshot 1</summary>
<img width="1440" alt="Screenshot 2023-05-15 at 3 26 39 AM" src="https://github.com/Expensify/App/assets/36869046/85ffc765-fce4-4c50-8616-d167930f550a">
</details> 

Then on the server, it is rounded to `1532330860146572300` 

<details><summary>Screenshot 2 </summary>
<img width="1440" alt="Screenshot 2023-05-15 at 3 26 56 AM" src="https://github.com/Expensify/App/assets/36869046/1d9c1aaa-f45a-498e-8693-5e92247e39eb">
</details> 

On front End , App fails to get the parentReportAction and displays report title as deleted
<details><summary>Screenshot 3</summary>
<img width="1440" alt="Screenshot 2023-05-15 at 3 25 55 AM" src="https://github.com/Expensify/App/assets/36869046/2e579ce3-62b5-4511-b953-29601cdab18c">
</details> 

I tried to add update Onyx and add the data from `1532330860146572254` to `1532330860146572300` and it was working. The title is updated as expected

<details><summary>Video</summary>

https://github.com/Expensify/App/assets/36869046/5848117d-7bd9-4657-ad3d-0b61bf5a4f54

</details>

Finnally after updating the report action , I am still unable to comment on the report

<details><summary>Screenshot 4</summary>

<img width="1440" alt="Screenshot 2023-05-15 at 5 06 12 AM" src="https://github.com/Expensify/App/assets/36869046/b252fe85-d054-4b13-a0af-d1b35d12a580">

</details> 


Other errors I got when trying to create new money request 

<img width="1009" alt="Screenshot 2023-05-15 at 2 47 26 AM" src="https://github.com/Expensify/App/assets/36869046/2f4fe5f8-c1b9-4676-85f8-e6a659ce9739">

mountiny committed:
Merge branch 'main' into bondy-thread-manual-requests

mountiny committed:
Access property safely

mountiny commented:
@fedirjh Thanks for the tests, the last failure has been resolved in the backend:

> Other errors I got when trying to create new money request

The other issues I cannot reproduce, I have tested locally again and I can create the reports just fine and access them, comment in them from the other account too. I think we should get this out. if @luacmartins can test this fine locally that would be great

mountiny commented on src/components/ReportActionItem/MoneyRequestAction.js:
```diff
@@ -67,15 +80,46 @@ const defaultProps = {
+        // If the childReportID is not present, we need to create a new thread
+        const childReportID = lodashGet(props.action, 'childReportID', '0');
+        if (childReportID === '0') {
+            const participants = _.uniq([props.session.email, props.action.actorEmail]);
```
In the expense report case 
<img width="714" alt="image" src="https://github.com/Expensify/App/assets/36083550/d6ead797-8217-4d31-b1e0-843dd6646c2f"> there is FAKE as a participant

I would keep it like this if it works dure to deadline

luacmartins replied:
Can we just filter those out?

mountiny committed:
Update spanish translation

mountiny committed:
Fix lint

luacmartins reviewed (commented):
Looking good! We'll also need to update the thread report to display the `MoneyRequestHeader` component

mountiny committed:
Merge branch 'main' into bondy-thread-manual-requests

mountiny commented:
Lovely, thanks!

trjExpensify commented:
Wahoo, thanks a lot @fedirjh! ❤️ We're hoping to get this one merged today to unblock a couple of other PRs for EC3. 🙏

fedirjh reviewed (approved):
LGTM and tests well.

mountiny commented:
@fedirjh thank you!

@luacmartins lets merge this and handle the rest in follow up PRs.

luacmartins reviewed (approved):
LGTM! As discussed, we need a followup to use the right report header for transaction threads

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/4983955649) to staging by https://github.com/luacmartins in version: 1.3.14-6 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/4997415254) to production by https://github.com/yuwenmemon in version: 1.3.14-14 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/5273496004) to staging by https://github.com/AndrewGable in version: 1.3.28-2 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5293001429) to production by https://github.com/AndrewGable in version: 1.3.28-5 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
